src/main/java/seedu/event/Event.java

```diff
@@ -109,6 +109,11 @@ public void setVenue(String venue) {
 
     @Override
     public String toString() {
-        return "Event: " + getName() + "at: " + getDatetime() + ", " + getVenue();
+        if (getDatetime() == "yyyy-MM-dd HHmm" && getVenue() == "") {
+            return "Event: " + getName();
+        } else if (getVenue() == "") {
+            return "Event: " + getName() + ", time: " + getDatetime();
+        }
+        return "Event: " + getName() + ", time: " + getDatetime() + ", venue: " + getVenue();
```

Comment:
Maybe have a string `output` that collects all the information (from name, datetime, to venue), then return `output`?

psuedocode:
`String output = "";`
`if (hasName()) appendNameToOutput();`
`if (hasDateTime()) appendDateTimeToOutput();`
`if (hasVenue()) appendVenueToOutput();`
`return output;`